class.jetpack-idc.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * This class will handle everything involved with fixing an Identity Crisis.
+ *
+ * @since 4.4.0
+ */
+class Jetpack_IDC {
+
+	/**
+	 * @var Jetpack_IDC
+	 **/
+	private static $instance = null;
+
+	/**
+	 * The wpcom value of the home URL
+	 * @var string
+	 */
+	static $wpcom_home_url;
+
+	static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new Jetpack_IDC;
+		}
+
+		return self::$instance;
+	}
+
+	private function __construct() {
+		if ( false === $urls_in_crisis = Jetpack::check_identity_crisis() ) {
+			return;
+		}
+
+		self::$wpcom_home_url = $urls_in_crisis['wpcom_home'];
+
+		add_action( 'admin_notices', array( $this, 'prepare_idc_notice' ) );
+	}
+
+	function prepare_idc_notice() {
+		if ( ! current_user_can( 'jetpack_disconnect' ) || ! Jetpack::is_active() || Jetpack::is_development_mode() ) {
+			return;
+		}
+
+		$this->enqueue_idc_notice_files();
+		$this->idc_notice_step_one();
+	}
+
+	/**
+	 * First "step" of the IDC mitigation. Will provide some messaging and two options/buttons.
+	 * "Confirm Staging" - Dismiss the notice and continue on with our lives in staging mode.
+	 * "Fix Jetpack Connection" - Will disconnect the site and start the mitigation...
+	 */
+	function idc_notice_step_one() {
+		$safe_mode_doc_link = 'https://jetpack.com/support/safe-mode';
+		$old_url = esc_url( self::$wpcom_home_url );
+		?>
+		<div class="jp-idc notice notice-warning">
+			<div class="jp-emblem">
+				<?php echo Jetpack::get_jp_emblem(); ?>
+			</div>
+			<p class="msg-top">
+				<?php esc_html_e( 'Jetpack Safe Mode.', 'jetpack' ); ?>
+			</p>
+			<hr />
+			<div class="msg-bottom-head">
+				<?php
+					_e(
+						sprintf(
+							'Jetpack has been placed into %1$sSafe Mode%2$s because we noticed this is an exact copy of %3$s.
+							Please confirm Safe Mode or fix the Jetpack connection. Select one of the options below or %1$slearn more about Safe Mode%2$s.',
+							'<a href="' . esc_url( $safe_mode_doc_link ) . '">',
+							'</a>',
+							'<a href="' . $old_url . '">' . $old_url . '</a>'
```

Comment:
It's escaped at the beginning of the function, but we could move the escaping to every sub-instance of it if you prefer